components/layout_2020/style_ext.rs

```diff
@@ -634,6 +634,20 @@ impl ComputedValuesExt for ComputedValues {
             return true;
         }
 
+        // From <https://www.w3.org/TR/css-will-change/#valdef-will-change-custom-ident>:
+        // > If any non-initial value of a property would create a stacking context on the element,
+        // > specifying that property in will-change must create a stacking context on the element.
+        if self.clone_will_change().bits.intersects(
+            WillChangeBits::STACKING_CONTEXT_UNCONDITIONAL |
+                WillChangeBits::TRANSFORM |
+                WillChangeBits::OPACITY |
+                WillChangeBits::PERSPECTIVE |
+                WillChangeBits::Z_INDEX |
```

Comment:
`z-index` by itself doesn't establish a stacking context, it needs `position`... but Blink and WebKit do it for `will-change: z-index` indeed. Gecko doesn't. I will file an issue to the CSSWG, for now it's fine either way.

Reply:
https://github.com/w3c/csswg-drafts/issues/11827